graviti/schema/package.py

```diff
@@ -287,12 +320,52 @@ def __getitem__(self, key: str) -> Type["PortexType"]:
 
         raise KeyError(key)
 
-    def __contain__(self, key: str) -> bool:
-        try:
-            self.__getitem__(key)
-            return True
-        except KeyError:
-            return False
+    def __setitem__(self, key: str, value: Type["PortexType"]) -> None:
+        type_ = self.get(key)
+        if type_:
+            if type_ != value:
+                raise KeyError("Duplicate names")
+            return
+
+        # TODO: Import "PortexExternalType" on toplevel will cause circular imports, to be solved.
+        # pylint: disable=import-outside-toplevel
+        from graviti.schema.template import PortexExternalType
+
+        if not issubclass(value, PortexExternalType):
+            raise TypeError("Local portex type is not supported yet")
+
+        package = value.package
+        subpackage = self._subpackages.get(package.repo)
+        if not subpackage:
+            subpackage = Subpackage(package)
+            self._subpackages[package.repo] = subpackage
+
+        subpackage[key] = value
+
+    def __iter__(self) -> Iterator[str]:
+        return chain(*self._subpackages.values())
+
+    def update(self, other: "Imports") -> None:
+        """Update the imports with another imports.
+
+        Arguments:
+            other: An :class:`Imports` instance whose types need to be updated to this imports.
+
+        """
+        assert not self._package and not other._package  # pylint: disable=protected-access
```

Comment:
Why can't they have packages?

Reply:
the base package in import is used for building extending package and local package, normally the these two process should not call this `update` method. I just add a simple check here to avoid unexpected error. I will remove the base class from Imports and use another class for package building in the future.